airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -502,32 +506,79 @@ def __init__(self,
         self.timeout = timeout
         self.metadata = metadata
         self.gcp_conn_id = gcp_conn_id
+        self.delete_on_error = delete_on_error
+
+    def _create_cluster(self, hook):
+        operation = hook.create_cluster(
+            project_id=self.project_id,
+            region=self.region,
+            cluster=self.cluster,
+            request_id=self.request_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        cluster = operation.result()
+        self.log.info("Cluster created.")
+        return cluster
+
+    def _delete_cluster(self, hook):
+        self.log.info("Deleting the cluster")
+        hook.delete_cluster(
+            region=self.region,
+            cluster_name=self.cluster_name,
+            project_id=self.project_id,
+        )
+        self.log.info("Cluster %s deleted", self.cluster_name)
```

Comment:
I wonder if it would not be a good idea to raise an exception here. It just seems weird to me that an operator that is supposed to create a cluster ends up deleting one instead and even returns a reference to no-longer-existing cluster. Raising an exception would also allow a second attempt to create the cluster on retry (if retries apply, of course).

Reply:
If the cluster isn't deleted, then the retries will fail because the cluster already exists (even if it exists in an ERROR state and is not usable).

Reply:
The ERROR cluster will eventually be lifecycled by dataproc, but in my experience that often takes longer (20-30 minutes) than typical retry counts / retry waits will cover.

Reply:
But I meant to raise an exception **after** the cluster is deleted. Then, as far as I understand, on retry we would follow the logic of cluster existing but being in 'DELETING' state, so we would get another chance to create it.

Reply:
Ah, yes, thank you!  +1

Reply:
Definitely agree! I've added exception 👌